### PR TITLE
add ut function for util/name.go

### PR DESCRIPTION
**Pull request by my-git9**
**What type of PR is this?**
add ut function for util/name.go

**What this PR does / why we need it**:

**Which issue(s) this PR about**:
https://github.com/karmada-io/karmada/issues/2579

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/util/names/names_test.go [hunk 1]
```diff
@@ -290,6 +290,26 @@ func TestGenerateDerivedServiceName(t *testing.T) {
...
 }
 
+func TestGenerateEstimatorDeploymentName(t *testing.T) {
+	tests := []struct {
+		name        string
+		clusterName string
+		expected    string
+	}{
+		{
+			name:        "",
```
Comment (Poor12) [inline pkg/util/names/names_test.go:300]: The name should be a description of the test case.

File: pkg/util/names/names_test.go [hunk 2]
```diff
@@ -369,3 +389,27 @@ func TestGenerateImpersonationSecretName(t *testing.T) {
...
+	tests := []struct {
+		name         string
+		namespace    string
+		resourcename string
+		gvk          string
+		expected     string
+	}{
+		{
+			name:         "test1",
+			namespace:    "namespaceA",
```
Comment (Poor12) [inline pkg/util/names/names_test.go:403]: For some names of example，I think "foo", "bar" and so on maybe better.

File: pkg/util/names/names_test.go [hunk 3]
```diff
@@ -369,3 +389,27 @@ func TestGenerateImpersonationSecretName(t *testing.T) {
...
+func TestGeneratePolicyName(t *testing.T) {
+	tests := []struct {
+		name         string
+		namespace    string
+		resourcename string
+		gvk          string
+		expected     string
+	}{
+		{
+			name:         "test1",
```
Comment (Poor12) [inline pkg/util/names/names_test.go:402]: name: "generate policy name"

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**